aten/src/ATen/native/cudnn/Conv.cpp

```diff
@@ -894,6 +900,52 @@ void raw_cudnn_convolution_forward_out(
     &zero, args.odesc.desc(), output.data_ptr()));
 }
 
+void raw_cudnn_convolution_forward_out(
+    const Tensor& output, const Tensor& input, const Tensor& weight,
+    IntArrayRef padding, IntArrayRef stride, IntArrayRef dilation, int64_t groups,
+    bool benchmark, bool deterministic) {
+  constexpr int64_t int_max = std::numeric_limits<int>::max();
+  const int64_t ni = input.numel();
+  const int64_t no = output.numel();
+  // if N * C * H * W <= int_max, then no need to split at all
+  if (ni <= int_max && no <= int_max) {
+    raw_cudnn_convolution_forward_out_32bit(output, input, weight, padding, stride, dilation, groups, benchmark, deterministic);
+    return;
+  }
+  // else, if C * H * W <= int_max, then we just need to split across the N dimension
+  //
+  // Here we use a simple heuristics to determine the size of each split
+  // We don't max out the 2^31 address space because this number is super
+  // large and very likely to get an OOM.
+  //
+  // TODO: we might need a more thorough test to find the best number here
+  constexpr int64_t max_worksize = 1024 * 1024 * 128;
```

Comment:
I think this can be safely increased, for 3d tensors this is likely too small.

Reply:
I will set this to, 1024 * 1024 * 512, does this sound good? 1024 * 1024 * 512 * fp32 = 2GB memory, if we have 2GB input + 2GB output + 2GB workspace = 6GB, this sounds like a good usage for a 16GB graphic card.

Reply:
Oh no, cuDNN seems to be using a huge workspace. Setting it to 1024 * 1024 * 512 will OOM on my card
```
Tried to allocate 8.25 GiB (GPU 0; 15.78 GiB total capacity; 4.09 GiB already allocated; 6.56 GiB free; 8.19 GiB reserved in total by PyTorch)
```
then 1024 * 1024 * 256 I think

Reply:
> Oh no, cuDNN seems to be using a huge workspace. Setting it to 1024 * 1024 * 512 will OOM on my card
> 
> ```
> Tried to allocate 8.25 GiB (GPU 0; 15.78 GiB total capacity; 4.09 GiB already allocated; 6.56 GiB free; 8.19 GiB reserved in total by PyTorch)
> ```
> 
> then 1024 * 1024 * 256 I think

You may check whether you can run on 32GB GPU memory.

Reply:
@wentaozhu Could you open a separate issue for that?